### ISSUE TITLE
Do our confidences mean anything? Reliability curves over FINAL emissions, per locale, on fresh holdout

### ISSUE THREAD

**Issue by GirlBossRush** [sister-software/mailwoman#1739]
Every confidence this system prints is uncalibrated, and we read them constantly — in traces, in gates, in this issue tracker. Before building anything that *consumes* uncertainty, measure whether the numbers mean what their magnitude implies.

## The specific reason to doubt them

The shipped decode distribution is not the trained one. `classifier.ts` composes the model's emissions with an additive prior stack before Viterbi runs:

```
emissions = addEmissionMatrix(emissions, fstPrior)            // #1497, default-ON
emissions = addEmissionMatrix(emissions, morphologyPrior)
emissions = addEmissionMatrix(emissions, trailingLocalityPrior)
… plus the placetype-pair delta, the placetype census delta, the conventions mask
```

So a per-token softmax read off the final emissions describes a distribution nothing trained against, and any calibration fitted at training time describes a distribution nothing evaluates. We also tune those priors on board deltas and have no instrument that would notice if a prior decalibrated the head.

Worked example from 2026-08-19, `Newport, Wales`: the model tagged `Wales` **B-locality at 0.68** and `Newport` **B-street at 0.83**. Both are wrong. Nothing tells us whether 0.68 is a fair 68% or a badly-scaled number, and the answer changes what a downstream consumer may do with it.

## The measurement

Reliability curves over FINAL emissions (post-bias-stack, the composition the resolver sees):

- bin per-token max-softmax into deciles, plot empirical tag accuracy per bin;
- report ECE and the reliability curve **per locale overlay** — the bias stack differs by overlay (`en-GB` ships a pair index and postcode anchor that `en-US` does not), so a single global curve would average away the thing we need;
- evaluate on a FRESH holdout draw (`{kind:"holdout", source:"fr"|"us"}`, seedless), never the gauntlet board: the board is a fixed adversarial set we actively optimise against, so a curve fitted on it describes the board;
- report separately for the tag classes that matter downstream (locality, region, street, venue), because the interesting miscalibration is per-class.

## Second leg, same instrument

Re-run the curve with the FST prior off (`gazetteer_prior: false`) and compare. That prices the question "does a default-ON additive prior decalibrate the head", which nothing currently asks. If it does, every future prior needs a calibration column beside its board delta.

## Why this is first

It is one eval and no mechanism. It either indicts the confidences we have been reading all along, or licenses them — and both #TBD-isotonic and #TBD-conformal are unfounded until it lands.

**Comment by GirlBossRush** [comment]
Measured. **The confidences do not mean what their magnitude implies, and the direction of the error reverses by tag** — so the number that looks reassuring in aggregate is the one hiding the problem.

Method: for every component the parse produced, the weakest per-token softmax over the tokens carrying that tag, against the golden v0.1.3 labels, using the harness's own `componentMatches` rule (a produced tag the gold row lacks counts as wrong). Script: `scratchpad/calibration-1739.ts`. Three legs — held-back TEST split (456 rows / 1,795 observations), the DEV tuning half (4,255 rows / 16,751), and TEST again with `gazetteer_prior: false`.

## The headline is a lie of averaging

| stratum | n | accuracy | ECE | direction |
| --- | --- | --- | --- | --- |
| ALL (test) | 1,795 | 0.868 | **0.036** | "we're fine" |
| ALL (dev) | 16,751 | 0.874 | 0.034 | same |
| `street` (dev) | 2,768 | **0.684** | **0.169** | **over**-confident |
| `locality` (dev) | 3,297 | 0.913 | 0.077 | **under**-confident |
| `region` (dev) | 2,714 | 0.955 | 0.089 | **under**-confident |
| locale `fr` (test) | 500 | 0.796 | 0.094 | over-confident at the top |
| locale `us` (test) | 1,259 | 0.894 | 0.049 | under-confident at the top |

A global ECE of 0.034 is what you would report doing this the obvious way, and it says nothing is wrong.

## What is actually wrong

**`street` is over-confident everywhere above 0.6, and it does not improve with confidence.** The dev curve, n=2,768:

```
  [0.6–0.7)  n=   79  conf=0.649  acc=0.456  gap=-0.193
  [0.7–0.8)  n=  130  conf=0.756  acc=0.615  gap=-0.141
  [0.8–0.9)  n=  368  conf=0.865  acc=0.723  gap=-0.142
  [0.9–1.0)  n= 1894  conf=0.926  acc=0.730  gap=-0.196
```

The model's most confident street spans are right **73%** of the time. 1,894 of 2,768 street observations sit in the top decile, so this is the modal case, not a tail.

**`locality` bends the other way**: top decile conf 0.932 → accuracy **0.996** (+0.064). Under-confident.

Both facts at once mean **temperature scaling cannot fix this**: one scalar moves both curves the same direction, and these need opposite corrections. Per-class calibration is required — the same Mondrian instinct as #1722, arriving from the model side.

**Confidences are compressed.** The top decile's MEAN is 0.92–0.93; the model essentially never says 0.99. Anything thresholding at 0.9 is thresholding inside the noise band, which is worth knowing before #1740 calibrates a 0.9 gate.

**Low confidence is not informative.** Test `[0.1–0.2)`: conf 0.131 → accuracy 0.571. Reading a 0.13 as "probably wrong" is wrong; it is a coin flip that leans right. (Consequence for last night's `Newport, Wales` triage: I read `Wales` tagged B-locality at 0.68 as a weak signal. At the measured curve, 0.68 for a locality is worth roughly 0.50–0.70 — weaker than the number looks, but the tag was still the modal outcome.)

## The two control legs

**The FST prior does NOT decalibrate the head.** TEST with `gazetteer_prior: false`: ALL ECE 0.034 vs 0.036, `street` 0.184 vs 0.183, `locality` 0.075 vs 0.077, accuracy 0.866 vs 0.868. Indistinguishable at this n. Whatever the prior does to ranking, it is not what is bending these curves — so the future rule "every new prior needs a calibration column" is worth keeping, but no existing prior is on the hook.

**Dev and test agree, so this is not overfitting to the tuning half.** `street` ECE 0.169 (dev) vs 0.183 (test); `locality` 0.077 vs 0.077; ALL 0.034 vs 0.036. The tuning half is not flattered relative to the held-back half — the miscalibration is a property of the model, not of our iteration.

## What this licenses and forbids

- **Forbids** reading raw confidences as probabilities anywhere they gate behaviour — including any future set-based mechanism (#1741) built on unadjusted softmax.
- **Licenses** #1740 and #1741, with a constraint neither had: calibration must be **per tag class** (and probably per locale — fr and us bend opposite ways), never a single global map.
- **Suggests** a separate question worth its own issue: `street` accuracy is 0.68 at ALL confidence levels. That is a model-quality finding this measurement stumbled on rather than a calibration one, and it is consistent with #1539 (suffix-boundary over-greed) — the street span is where our tagging is genuinely weakest.

Not doing yet: fitting the per-class calibrators. This issue asked whether the numbers mean anything; they do not, differently per class, and that is the deliverable.

**Comment by GirlBossRush** [comment]
Closing — the measurement is delivered and the follow-ups have their own homes.

## The answer

**No, not uniformly — and the aggregate is what hides it.** Global ECE 0.036 reads as calibrated;
underneath, the error reverses direction by class:

| stratum (dev, n=16,751) | accuracy | ECE | direction |
| --- | --- | --- | --- |
| `street` | **0.684** | **0.169** | over-confident |
| `locality` | 0.913 | 0.077 | under-confident |
| `region` | 0.955 | 0.089 | under-confident |
| ALL | 0.874 | 0.034 | the average of two opposite errors |

The `street` curve is the sharp one and it is the modal case, not a tail: 1,894 of 2,768 observations
sit in the top decile, where mean confidence 0.926 buys accuracy **0.730**.

**One scalar temperature moves both curves the same way**, so it cannot fix either — per-class
calibration is forced. That is the Mondrian instinct arriving from the model side rather than the
diagnosis side, and it is now cited in #1722 as measured support for the class-conditional design.

Two secondary facts that changed how traces get read: confidences are **compressed** (top-decile mean
0.92–0.93; the model essentially never says 0.99), so a 0.9 gate thresholds inside the noise — a
constraint #1740 inherited. And **low confidence carries little information**: the `[0.1–0.2)` bin is
right 57% of the time.

Both control legs came back clean. `gazetteer_prior: false` gives ECE 0.034 vs 0.036 — no shipped prior
is on the hook. Dev and test agree (street 0.169 vs 0.183), so this is the model's property and a
calibrator fitted on held-out data should transfer.

## Where the work went

- **#1740** — measured the coarse placer's own surface: ECE **0.001** over 145,629 held-out rows, so isotonic there is a no-op. Closed negative, with the real finding being that 59% of errors admitted at the 0.9 gate come from two reciprocal pairs (CN↔TW, ES↔IT) that no scalar can separate.
- **#1722** — the per-class evidence, plus the corrected conformal framing.
- **#1539** — `street` accuracy is 0.68 at EVERY confidence level, which is a model-quality finding rather than a calibration one and belongs with the suffix-boundary work.

## Instrument

`scripts/diagnostic/calibration-reliability.ts` (5ea1dc784), kept and tracked. It takes a corpus and a
label, so re-running it against the next model is one command rather than a rebuild.

One thing this issue asked that is still unanswered, recorded rather than quietly dropped: nothing here
measured the RESOLVER's confidence surfaces (importance, the fuzzy tier's score floor). #1740 named them
as wanting the same treatment and neither has been touched.
